privacy/anonymizing-your-mac-address.md

```diff
@@ -170,3 +172,48 @@ To disable MAC Randomizing if you find that a network connecting to does not lik
 - Also in `fedora-23` type `sudo systemctl disable macspoof@enp0s0`
 - Remove the service file `sudo rm /etc/systemd/system/macspoof@.service` in TemplateVM
 - Delete the package `sudo dnf remove macchanger`
+
+Method 2
+========
+
+An alternative approach is to use udev rules.
+
+This method does not require you to specify interface names, will apply to all interfaces, even those added after boot, and will retain the random MAC address through sleep/wake cycles.
+
+Install **macchanger** package in the TemplateVM backing sys-net.
+
+In that template create a udev rule using an editor of your choice.
+
+In /etc/udev/rules.d create the file 00-mac-spoof.rules with content:
+```
+SUBSYSTEM=="net", ACTION=="add", ATTR{type}=="1", RUN+="/usr/bin/macchanger -e  $name"
```

Comment:
To not mess with other configuration. For example `vif*` interfaces are excluded from NetworkManager using mac address (the only supported option): https://github.com/QubesOS/qubes-core-agent-linux/blob/master/network/network-manager-prepare-conf-dir#L11-L16
Otherwise NetworkManager would try to run DHCP client on downlinks to the VMs...
As for `eth0` in VMs (uplink to netvm), there is already setting for that (available via `qvm-prefs`), so users may expect to have this setting respected. In practice this is much more useful for HVM (and Windows there, to not complain about different MAC than it was installed on - in case of TemplateVM and TemplateBasedVMs). But still the setting is there.